pkg/generate/config/config.go

```diff
@@ -133,19 +133,22 @@ type ResourceConfig struct {
 
 	// SpecFields is a list of instructions about additional Spec fields
 	// on this Resource
-	SpecFields []*SpecFieldConfig `json:"spec_fields"`
+	SpecFields []*AdditionalFieldConfig `json:"spec_fields"`
+	// StatusFields is a list of instructions about additional Status fields
+	// on this Resource
+	StatusFields []*AdditionalFieldConfig `json:"status_fields"`
 }
 
-// SpecFieldConfig instructs the code generator how to handle an additional
-// field in the Resource's SpecFields collection. This additional field can source
-// its value from a shape in a different API Operation.
-type SpecFieldConfig struct {
+// AdditionalFieldConfig instructs the code generator how to handle an additional
+// field in the Resource's SpecFields/StatusFields collection. This additional field
+// can source its value from a shape in a different API Operation.
+type AdditionalFieldConfig struct {
```

Comment:
eventually, we'll probably want to consolidate the existing `FieldConfig` with this, but this is cool for now :)